mobile/src/components/BookFilter.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState } from 'react';
+import { View, TextInput, StyleSheet, Keyboard, Dimensions, Image, Text } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import { Language, Languages } from '../models/Languages';
+import { TextStyles } from '../styles/TextStyles';
+import { Colors } from '../styles/Colors';
+
+const { width } = Dimensions.get('window');
+
+// ISO 639-1 2-letter language codes (from the Languages file) and the number of languages
+const languages = Object.keys(Languages) as Language[];
+
+type BookFilterProps = { onFilter: (text: string, langs: Language[]) => void };
+
+/**
+ * Component that renders a search bar for the user to find books by title or author.
+ */
+export const BookFilter: React.FC<BookFilterProps> = ({ onFilter }) => {
+  // the search term, the language dropdown's visibility, and the set of selected languages
+  const [search, setSearch] = useState('');
+  const [dropdownVisible, setDropDownVisible] = useState(false);
+  const [langs, setLangs] = useState(new Set<Language>(['en']));
+
+  // toggles the languages in the language dropdown and re-filters the books
+  const toggleLang = (lang: Language): void => {
+    const newLangs = new Set<Language>(langs);
+    if (langs.has(lang)) {
+      newLangs.delete(lang);
+    } else {
+      newLangs.add(lang);
+    }
+
+    setLangs(newLangs);
+    onFilter(search, Array.from(newLangs));
+  };
+
+  // re-filters the books when the search term is changed
+  const searchBooks = (text: string): void => {
+    setSearch(text);
+    onFilter(text, Array.from(langs));
+  };
+
+  // the language dropdown component
+  const dropdown = (
+
+    <View style={styles.dropdown}>
+
+      {languages.map((lang: Language) => (
+        <TouchableOpacity
+          key={lang}
+          onPress={() => toggleLang(lang)}
+        >
+
+          <View style={styles.nameBoxContainer}>
+
+            <Text style={{ ...TextStyles.caption4, alignSelf: 'center' }}>{Languages[lang]}</Text>
+
+            <View style={styles.box}>
+              {langs.has(lang) ?
+                <Image style={styles.boxChecked} source={require('../../assets/images/check-square-solid.png')}/>
+                :
+                null}
```

Comment:
This is fine, just a tip that this can be reduced to `langs.has(lang) && <Image ...>`